marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/steps/MappingStepController.java

```diff
@@ -83,6 +84,20 @@
         return new ResponseEntity<>(body, headers, HttpStatus.OK);
     }
 
+    @RequestMapping(value = "/{stepName}/newDoc", method = RequestMethod.GET)
+    @ApiOperation(value = "Returns an XML or JSON source document as a string of JSON")
+    @Secured("ROLE_readMapping")
+    public ResponseEntity<JsonNode> getNewDocument(@PathVariable String stepName, @PathVariable String docUri) {
+        // Step defines which database to pull the identified doc from.
+        final JsonNode stepNode = newService().getStep(STEP_DEFINITION_TYPE, stepName);
```

Comment:
Should we determine the database from the step name here or on the backend, within `getNewDocument.sjs`?  The latter has the advantage of not presuming staging or final ...it's whatever the sourceDatabase property is.  From `getDocument.sjs`: `{database: xdmp.database(mappingStep.sourceDatabase)}`

Reply:
Yeah definitely have the backend figure it out.

Reply:
Addressed in fork; will be part of next or updated PR.